test/update-bundler-manifest.js

```diff
@@ -116,19 +133,28 @@ async function fetchTestResults() {
 
   const filePath = path.join(
     os.tmpdir(),
-    `next-test-results.${Math.floor(Math.random() * 1000).toString(16)}.zip`
+    `next-test-results.${Math.floor(Math.random() * 100000).toString(16)}.zip`
   )
 
-  subprocess.stdout.pipe(fs.createWriteStream(filePath))
+  let file
+  try {
+    file = await fs.open(filePath, 'w')
+
+    subprocess.stdout.pipe(file.createWriteStream())
 
-  await subprocess
+    await subprocess
+  } finally {
+    await file.close()
+  }
```

Comment:
I was seeing some frequent flaky errors in CI that seemed related to partially written zip files.

As far as I can tell, if you use `fs.createWriteStream()`, you need to make sure the file finishes opening before you can use the stream (e.g. https://github.com/sindresorhus/execa/blob/99d1741d2525eca71b986282148bbf2983356428/docs/streams.md#output).

Hopefully this fixes it. There's also some possibility this needs to be flushed (`close` does not flush by default), but I'm going to assume that `execa` flushes the file, as I can't find anything about that in the `execa` docs (and that would be the sane thing for them to do).